### ISSUE TITLE
Tables need a better way to size / re-size column widths

### ISSUE THREAD

**Issue by jdanish** [netcreateorg/netcreate-itest#108]
Ideally we'd space them out and then have a drag to change, or less ideally we'd specify in the template.

@benloh This can wait until after the major functionality is in, but I wanted to make sure I didn't forget to stick it on a list.

**Comment by benloh** [comment]
![image](https://github.com/netcreateorg/netcreate-itest/assets/1403057/48c0ca1d-fbe0-4c67-b8c9-9e439ad9e7b9)

**Comment by benloh** [comment]
Joshua wrote:

My worry on the table is that on smaller screen kid-machines we are already too wide, so we need to figure out *something*. I had been thinking at a minimum we want the option to re-size columns. Had also been thinking that the approach some pro spreadsheets use where you can do a popup with a check-box stop hide / show some columns might make sense, with a way of setting defaults. But now I am wondering if that is "too much" for many kids? 

Another worry on collapsing the provenance into one column and using a popup would be if we need / want to sort by any of the specific fields. If we don't need to sort the table that way, no big deal. If we do ...  Otherwise, no problem. 

One thought I just had is what if we do let kids resize columns, but handle crunched columns more elegantly. For example, instead of having "Designer" overrun as it does below, maybe we just start truncating things so it is Des... and that's not helpful, but you can choose which ones do that or not?  (I'm imagining already the performance hit too)

<img width="926" alt="Screenshot 2024-05-18 at 9 15 34 PM" src="https://github.com/netcreateorg/netcreate-itest/assets/5529898/d2887da7-770d-4917-b24d-2f30fe04f623">

If you want to mockup with something to keep moving I'm all for it, and if you have ideas cool. Otherwise I think I'll need to do some googling and thinking and brainstorming on this. Thoughts?




For reference, just found this: https://www.pencilandpaper.io/articles/ux-pattern-analysis-enterprise-data-tables

TLDR, some ideas:
Allow freeze, hide, reorder of columns and horizontal scrolling in various combinations.
Freezing the key (non-optional) columns and then allowing horizontal scrolling on the others might be easiest?  Similar to this image but with us deciding what is frozen: https://assets-global.website-files.com/65d605a3b4417479c154329f/65f02505add6d3eda2bfa903_Table-23-Freeze-Dropdown.svg
